<?xml version="1.0"?>
<launch>
    <arg name="robot_name" default="my_gen3"/>
    <arg name="tracking_mode" default="toggle"/>
    <arg name="controller_side" default="right"/>
    <arg name="compensate_orientation" default="true"/>

    <!-- Oculus controller feedback: -->
    <include file="$(find oculus_ros)/launch/controller_feedback.launch">
        <arg name="controller_side" value="$(arg controller_side)"/>
    </include>

    <!-- Oculus mapping: -->
    <group ns="/$(arg robot_name)">
        <node pkg="kinova_positional_control" type="oculus_mapping.py" name="oculus_mapping" output="screen">
            <param name="robot_name" type="string" value="$(arg robot_name)"/>
            <param name="controller_side" type="string" value="$(arg controller_side)"/>
        </node>
    </group>

    <!-- Teleoperation module: -->
    <include file="$(find kinova_positional_control)/launch/teleoperation.launch">
        <arg name="robot_name" value="$(arg robot_name)"/>
        <arg name="tracking_mode" value="$(arg tracking_mode)"/>
        <arg name="compensate_orientation" value="$(arg compensate_orientation)"/>
    </include>
</launch>